diff --git a/source/effect_codegen_hlsl.cpp b/source/effect_codegen_hlsl.cpp
index 2e9307ab..10862888 100644
--- a/source/effect_codegen_hlsl.cpp
+++ b/source/effect_codegen_hlsl.cpp
@@ -1871,7 +1871,7 @@ protected:
 		if (flags & 0x1)
 			attributes += "[unroll] ";
 		if (flags & 0x2)
-			attributes += _shader_model >= 40 ? "[fastopt] " : "[loop] ";
+			attributes += "[loop] ";
 
 		// Condition value can be missing in infinite loop constructs like "for (;;)"
 		std::string condition_name = condition_value != 0 ? id_to_name(condition_value) : "true";
